src/main/java/com/example/tournamentMaker/randomResultGenerator/RandomResultService.java

```diff
@@ -121,10 +94,47 @@ private void drawFootballRoundResult(int roundNumber, Tournament tournament) {
         });
     }
 
-    private int findGoalKeeperJerseyNumber(Team team) {
+    private void updateSpecificStatistics(int hostPoints, int guestPoints, FootballStatistics hostStatistics, FootballStatistics guestStatistics, Team host, Team guest) {
+        List<Integer> hostJerseyNumbersGoalScorers = drawLotJerseyNumbersGoalScorers(hostPoints, host);
+        List<Integer> guestJerseyNumbersGoalScorers = drawLotJerseyNumbersGoalScorers(guestPoints, guest);
+        List<Integer> hostJerseyNumbersWithYellowCard = drawLotUniqueJerseyNumbers(RandomUtil.generateRandomNumber
+                (MIN_YELLOW_CARDS_IN_MATCH, MAX_YELLOW_CARDS_IN_MATCH), host);
+        List<Integer> guestJerseyNumbersWithYellowCard = drawLotUniqueJerseyNumbers(RandomUtil.generateRandomNumber
+                (MIN_YELLOW_CARDS_IN_MATCH, MAX_YELLOW_CARDS_IN_MATCH), guest);
+        List<Integer> hostJerseyNumbersWithRedCard = drawLotUniqueJerseyNumbers(RandomUtil.getRandomRedCardCount(), host);
+        List<Integer> guestJerseyNumberWithRedCard = drawLotUniqueJerseyNumbers(RandomUtil.getRandomRedCardCount(), guest);
+
+        updateStatisticsForTeam(hostStatistics, host, hostJerseyNumbersGoalScorers,
+                hostJerseyNumbersWithYellowCard, hostJerseyNumbersWithRedCard);
+        updateStatisticsForTeam(guestStatistics, guest, guestJerseyNumbersGoalScorers,
+                guestJerseyNumbersWithYellowCard, guestJerseyNumberWithRedCard);
+
+        if (teamScoredZeroGoals(hostPoints)) {
+            resultService.updateSpecificStatisticInTeam(List.of(findJerseyNumberPlayerInSpecificPosition
+                    (guest, FootballPosition.GOALKEEPER)), guest, guestStatistics.getPlayersIdWithCleanSheets());
+        }
+        if (teamScoredZeroGoals(guestPoints)) {
+            resultService.updateSpecificStatisticInTeam(List.of(findJerseyNumberPlayerInSpecificPosition
+                    (host, FootballPosition.GOALKEEPER)), host, hostStatistics.getPlayersIdWithCleanSheets());
+        }
```

Comment:
to tez troche smierdzi , jakbym tego nie przeczytal 3x to bym nie zauwazyl co sie rozni